### ISSUE TITLE
Parallelization of FirefoxDriver in v2.13.0+ of Selenium causes socket lock

### ISSUE THREAD

**Issue by yurodivuie** [dev9com/parallel-webtest#4]
At present, when running multiple threads using FirefoxDriver, I'm seeing intermittent failures to retrieve a webdriver from the WebDriverLauncher.getNewWebDriverInstance.  This appears to be the sequence of events:
1. WebDriverLauncher.getNewWebDriverInstance is called @Before a test
2. Untrapped WebDriverException is generated, and execution seems to stop for this test.
3. The next test method in the same class begins execution, and WebDriverLauncher.getNewWebDriverInstance is called again, since a driver for the class is not successfully created by the first test method.
4. Tests continue to execute without issue, but the method count is off (the @after method fails when it encounters a null driver), so the driver is never closed.

See this article for details: http://code.google.com/p/selenium/wiki/ScalingWebDriver

I'm still researching... may put in an exception handler that retries browser creation (or reduce threads) as a temporary fix...

**Comment by yurodivuie** [comment]
Note: this only affects local test runs using a local browser with multiple threads.

**Comment by yurodivuie** [comment]
The temporary fix needs to be tested on windows; there's a new branch called "issue-4" that will retry creating a local browser if a WebDriverException is generated.  There's a set of 20 "swarm" tests that I'm using to try to trigger the port lock, but I've had not success on the Mac.

**Comment by yurodivuie** [comment]
Tested issue-4 branch in windows.  Using my mediocre home PC, I could run ten tests simultaneously with the fix (passing through 20 test classes with two methods each, using Firefox locally against a public site).  Judging by the error logs, four failed driver creations were successfully recovered from.  If I try to run twenty threads simultaneously, the number of errors overwhelms the retry count (set to 3), and tests start failing again.

The fix mitigates the problem by retrying when creating local browsers, using the same retry block that is used for remote web driver provisioning.  I also added logging that advises the user to tone down the number of threads used for parallelization if this is encountered.  I've merged this branch back into master... which apparently makes it vanish, which I was not aware of.
